app/index.py

```diff
@@ -11,19 +10,50 @@
 app.secret_key = get_config().app_secret_key
 
 
+def authenticated(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if 'authenticated' not in session:
+            return redirect(url_for('login'))
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
 @app.route('/')
 @app.route('/index')
 def index():
     return render_template('base.html')
 
 
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    if request.method == 'POST':
+        if request.form['key'] == get_config().api_key:
+            session['authenticated'] = True
+            return redirect(url_for('view_all_data'))
+        else:
+            error_msg = 'Invalid password'
+        return render_template('login.html', error=error_msg)
+    return render_template('login.html')
```

Comment:
To powinny być 2 endpointy. Jeden z metodą GET, a drugi z metodą POST. Jeden kontroler jedna funkcjonalność

Reply:
Rozdzielone na "view_login_form" i "login" w `app.domain.common.authentication.routes`